feat(WarningModal): add text confirmation option (#540)
patternfly/react-component-groups: packages/module/patternfly-docs/content/extensions/component-groups/examples/WarningModal/WarningModal.md

```

### Warning modal with a text confirmation

> To confirm that a user wishes to initiate a selected action, you can add a text confirmation.

Reviewer: ```suggestion
To ensure the user intentionally initiates a selected action, you can add a confirmation text input.
```
